server/boot/customRoutes.js

```diff
@@ -1,11 +1,42 @@
-var loopback = require("loopback");
-module.exports = function(app) {
-  var router = app.loopback.Router();
-  var Stores = app.models.Stores;
-  router.get("/api/stores/location", function(req, res) {
-    var location = req.query.location;
-    var lat = req.query.lat;
-    var lng = req.query.lng;
+const loopback = require("loopback");
+module.exports = function (app) {
+  const router = app.loopback.Router();
+  const Stores = app.models.Stores;
+  const Bookings = app.models.Bookings;
+  const Stores_slots = app.models.stores_slots;
+  const bodyParser = require("body-parser");
+  app.use(bodyParser.json({ extended: true }));
+
+  router.post("/api/bookings", (req, res) => {
+    //front-end req needs to pass in store_id, slot_id, user_id
+    //also takes order_details for users with email-linked accounts
+    const newBooking = req.body;
+    const store = Stores.findById(newBooking.store_id);
+    const bookings = Bookings.find({ "where": { "slot_id": newBooking.slot_id } });
```

Comment:
You could have used include filter here.  which would have reduced ons of the query.

Reply:
Changed it so I'm only using 2 queries.